### PR TITLE
[Exposing ALTS Context 2/2] Utility Wrapper Class

### PR DESCRIPTION
**Pull request by ZhenLian**
This PR serves the same purpose with the reverted PR: https://github.com/grpc/grpc/pull/21456.
We plan to manually provide a patch to avoid breaking internal build after this PR is merged.

### CODE REVIEW

**Comment by ZhenLian** [comment]
@jiangtaoli2016 Could you please take a look? This is basically the same as the reverted one, with some nit change; thanks in advance!

File: include/grpcpp/alts_context.h [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+  grpc::string application_protocol_;
+  grpc::string record_protocol_;
+  grpc::string peer_service_account_;
+  grpc::string local_service_account_;
+  grpc_security_level security_level_ = GRPC_SECURITY_NONE;
+  RpcProtocolVersions peer_rpc_versions_ = {{0, 0}, {0, 0}};
+};
+
+// GetAltsContextFromAuthContext helps to get the AltsContext from AuthContext.
+// Please make sure the underlying protocol is ALTS before calling this
```
Comment (jiangtaoli2016) [inline include/grpcpp/alts_context.h]: nit. I would rephrase as

"If ALTS is not the transport security protocol used to establish the connection, this function will return nullptr."

**Comment by jiangtaoli2016** [comment]
By the way, in the future, you shall revert the reverted PR, so that reviewer can easily see what has been modified from the previous PR.

**Comment by ZhenLian** [comment]
@jiangtaoli2016 
Done. Thanks for pointing this out!

**Comment by jiangtaoli2016** [comment]
PR https://github.com/grpc/grpc/pull/21547 will replace this PR.
